src/Symfony/Component/AssetMapper/CacheWarmer/AssetMapperCacheWarmer.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\AssetMapper\CacheWarmer;
+
+use Symfony\Component\AssetMapper\AssetMapper;
+use Symfony\Component\AssetMapper\AssetMapperInterface;
+use Symfony\Component\AssetMapper\CompiledAssetMapperConfigReader;
+use Symfony\Component\AssetMapper\Event\PreAssetsCompileEvent;
+use Symfony\Component\AssetMapper\ImportMap\ImportMapGenerator;
+use Symfony\Component\AssetMapper\Path\PublicAssetsFilesystemInterface;
+use Symfony\Component\Console\Output\NullOutput;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\HttpKernel\CacheWarmer\CacheWarmerInterface;
+
+/**
+ * @author Ryan Weaver <ryan@symfonycasts.com>
+ * @author Alexandre Daubois <alex.daubois@gmail.com>
+ */
+final class AssetMapperCacheWarmer implements CacheWarmerInterface
+{
+    public function __construct(
+        private readonly CompiledAssetMapperConfigReader $compiledConfigReader,
+        private readonly AssetMapperInterface $assetMapper,
+        private readonly ImportMapGenerator $importMapGenerator,
+        private readonly PublicAssetsFilesystemInterface $assetsFilesystem,
+        private readonly ?EventDispatcherInterface $eventDispatcher = null,
+    ) {
+    }
+
+    public function isOptional(): bool
+    {
+        return false;
+    }
+
+    public function warmUp(string $cacheDir, ?string $buildDir = null): array
+    {
+        $this->compiledConfigReader->removeConfig(AssetMapper::MANIFEST_FILE_NAME);
+        $this->compiledConfigReader->removeConfig(ImportMapGenerator::IMPORT_MAP_CACHE_FILENAME);
```

Comment:
How can anyone disable this cache warmer ? As it destroys data and people have many reasons to separate cache warm and asset build, this feels more than required

(sorry if i missread something here)

Reply:
I think you have better knowledge of this component than me 🙂 would a container parameter do the job? Something like `asset_mapper.enable_cache_warmer: true`

Reply:
I would make it optional to start, so that cache:clear does not force all assets (images, videos, idk.. to be read in memory).

And i realize this is the reason for this architeccture at first. I remember asking about the config files beeing outside, and i think the answer was to make this independant from the php code. 

So, we _**must**_ have a way (even, the default way i'd say for BC reasons) to build a Symfony app without deleting all its asset config.